### ISSUE TITLE
[svg-native] support xlink:href on gradients

### ISSUE THREAD

**Issue by dirkschulze** [w3c/svgwg#670]
The `xlink:href` attribute on `<linearGradient>` or `<radialGradient>` is ignored right now. We should consider keeping it.

Without support for `objectBoundingBox`, referencing other gradients is straight forward.

The `xlink:href` is mainly used to share color stops between multiple gradients. W/o `objectBoundingBox` this is even more important. Otherwise SVG files blow up a lot.

**Comment by jarek-foksa** [comment]
> The xlink:href is mainly used to share color stops between multiple gradients. W/o objectBoundingBox this is even more important. Otherwise SVG files blow up a lot.

This shouldn't be a problem if you compress the file. In my opinion gradient inheritance is a feature that makes sense in a rich document format (such as SVG), not in an output format (such as SVG Native).


So the only advantage of this 
If the same gradient stops are used by say 100 different objects

Is there any authoring tool that would actually take advantage of gradient inheritance? Most SVG files I have seen 
From my experience most SVG files don't make extensive use of gradient inheritance

**Comment by litherum** [comment]
> Otherwise SVG files blow up a lot.

Is this the main motivation? If so, we should compare file sizes of compressed-without-stop-sharing files against compressed-with-stop-sharing files.

**Comment by dirkschulze** [comment]
@litherum fair enough. That assumes that compression will happen though. You are probably just thinking about sending SVGs over the wire or as embedded vector for OT fonts. Can we make this assumption on native apps? Usually, applications don't compress SVG files upon shipping. Decompression takes time and native applications come with a lot of icons. Adobe's application may have more than thousand icons for one application. Decompression does add up on so many icons.

**Comment by jarek-foksa** [comment]
@dirkschulze Aren't all Adobe products using flat icons for the UI? I think this is also the case for the majority of iOS, Android, macOS and Windows 10 apps that follow the Human Interface Guidelines.

**Comment by dirkschulze** [comment]
@jarek-foksa I am not sure what you are referring to with flat icons in UI? Do you ask if we support gradients in icons used within Adobe products? We aim to support SVG Native features in our UI.

**Comment by jarek-foksa** [comment]
@dirkschulze I meant that most modern apps are adopting flat design patterns which are discouraging heavy use of gradients. Therefore, the potential benefits of supporting gradient reuse would be negligible even for an application that defines over a thousand of icons such as Photoshop.

**Comment by dirkschulze** [comment]
@jarek-foksa Design patterns come and go :)
